src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts

```diff
@@ -125,18 +132,23 @@ generateData() {
   /***
    *   Create colletion creates the content Id and navigate
    * */
-  createCollection() {
+  createCollection(modalCollection) {
+    this.showLoader = true;
    const requestData = {
       content: this.generateData()
     };
+    this.loaderMessage = {
+      'loaderMessage': this.resourceService.messages.stmsg.m0016,
+  };
     this.editorService.create(requestData).subscribe(res => {
       this.showLoader = false;
-      const type = 'CollectionEditor';
-      const state = '';
+      const type = 'Collection';
+      const state = 'state';
       const framework = 'framework';
+      modalCollection.approve();
```

Comment:
why we need to close model, angular will destroy this model when we navigate to different url

Reply:
Modal remains opened even after we navigate to other router. so we have to close modal, also i am displaying the loader message.